crates/duty-tracker/src/contract_manager.rs

```diff
@@ -80,7 +84,24 @@ impl ContractManager {
                 }
             };
 
-            // TODO(proofofkeags): synchronize state with chain state
+            let current = match rpc_client.get_block_count().await {
+                Ok(a) => a,
+                Err(e) => {
+                    crash(e.into());
```

Comment:
I see that this `crash` function is still a TODO.
What if we define it somewhere else that can be used throughout the codebase.
Maybe if you are up to it we can define it in some utils crate or primitive.

Reply:
Agreed. What error type should it accept?

Reply:
Can be an error log macro then a panic macro. We must be sure to log the error before panicking

Reply:
I don't mind a panic macro except that it can be overwhelming in the output for operators. I'd imagine it could be useful to generate the error report to a file and then provide a small summary to stderr. But we can definitely start with a panic for now and refine later.